BitFaster.Caching/Lru/CustomPolicy.cs

```diff
@@ -0,0 +1,293 @@
+﻿using System;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+
+namespace BitFaster.Caching.Lru
+{
+    /// <summary>
+    /// Defines a policy for determining the expiry of an item.
+    /// </summary>
+    public interface IExpiry<K, V>
+    {
+        /// <summary>
+        /// Gets the expiry for an item after it is created.
+        /// </summary>
+        Func<K, V, TimeSpan> GetExpireAfterCreate { get; }
+
+        /// <summary>
+        /// Gets the expiry for an item after it is read.
+        /// </summary>
+        Func<K, V, TimeSpan> GetExpireAfterRead { get; }

```

Comment:
This cannot accept the existing TTL as input, because we have converted it to ticks since the machine started which is non-reversable.